FWCore/Services/plugins/InitRootHandlers.cc

```diff
@@ -217,49 +216,60 @@ namespace {
        && (el_message.find("fill branch") != std::string::npos)
        && (el_message.find("address") != std::string::npos)
        && (el_message.find("not set") != std::string::npos)) {
-        el_severity = SeverityLevel::kFatal;
+        el_severity = edm::RootHandlers::SeverityLevel::kFatal;
       }
 
       if ((el_message.find("Tree branches") != std::string::npos)
        && (el_message.find("different numbers of entries") != std::string::npos)) {
-        el_severity = SeverityLevel::kFatal;
+        el_severity = edm::RootHandlers::SeverityLevel::kFatal;
       }
 
 
   // Intercept some messages and downgrade the severity
 
-      if ((el_message.find("no dictionary for class") != std::string::npos) ||
-          (el_message.find("already in TClassTable") != std::string::npos) ||
-          (el_message.find("matrix not positive definite") != std::string::npos) ||
-          (el_message.find("not a TStreamerInfo object") != std::string::npos) ||
-          (el_message.find("Problems declaring payload") != std::string::npos) ||
-          (el_message.find("Announced number of args different from the real number of argument passed") != std::string::npos) || // Always printed if gDebug>0 - regardless of whether warning message is real.
-          (el_location.find("Fit") != std::string::npos) ||
-          (el_location.find("TDecompChol::Solve") != std::string::npos) ||
-          (el_location.find("THistPainter::PaintInit") != std::string::npos) ||
-          (el_location.find("TUnixSystem::SetDisplay") != std::string::npos) ||
-          (el_location.find("TGClient::GetFontByName") != std::string::npos) ||
-          (el_location.find("Inverter::Dinv") != std::string::npos) ||
-          (el_message.find("nbins is <=0 - set to nbins = 1") != std::string::npos) ||
-          (el_message.find("nbinsy is <=0 - set to nbinsy = 1") != std::string::npos) ||
-          (level < kError and
-           (el_location.find("CINTTypedefBuilder::Setup")!= std::string::npos) and
-           (el_message.find("possible entries are in use!") != std::string::npos))) {
-        el_severity = SeverityLevel::kInfo;
+      std::vector<std::string> in_message = {
```

Comment:
This is requiring the construction of this vector each time RootErrorHandlerImpl is called. Move this outside of the function and declare it `const`.
Do this for all the `std::vector<std::string>`.

Reply:
In addition, better to change this to `const std::vector<const char* const>` since no memory copies would then be needed to create `std::string`.

Reply:
If I use direct list initialization instead of copy list initialization, is it still necessary to use char arrays?

Reply:
Heck, it might be possible to completely do away with any allocations by using `constexpr std::array<const char* const, ...>`

Reply:
To follow the standard, the system has to malloc both the std::vector and all the std::strings each call.

Reply:
But moving them outside of the function should avoid that, right? Does direct list initialization still entail memory copies?